openedx/core/djangoapps/theming/management/commands/create_sites_and_configurations.py

```diff
@@ -75,26 +73,23 @@ def _create_oauth2_client(self, url, site_name, service_name, service_user):
             service_name=service_name,
             site_name="" if site_name == "edx" else "-{}".format(site_name)
         )
-        client, created = Client.objects.update_or_create(
+        Application.objects.update_or_create(
             client_id=client_id,
             defaults={
                 "user": service_user,
                 "name": "{site_name}_{service_name}_client".format(
                     site_name=site_name,
                     service_name=service_name,
                 ),
-                "url": url,
                 "client_secret": "{service_name}-secret".format(
                     service_name=service_name
                 ),
-                "client_type": CONFIDENTIAL,
-                "redirect_uri": "{url}complete/edx-oidc/".format(url=url),
-                "logout_uri": "{url}logout/".format(url=url)
+                "client_type": Application.CLIENT_CONFIDENTIAL,
+                "authorization_grant_type": Application.GRANT_PASSWORD,
```

Comment:
What made you choose GRANT_PASSWORD?  I think this might be "Authorization Code" if this is for SSO.

Reply:
This was the default in `DOTAdapter`'s `create_client`, so I just stuck with that. I'm happy to change it if needed.